lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -4062,6 +4062,101 @@ LogicalResult AtenPermuteOp::verify() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// AtenLinalgCrossOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult AtenLinalgCrossOp::verify() {
+
+  auto selfType = getSelf().getType().cast<BaseTensorType>();
+  auto otherType = getOther().getType().cast<BaseTensorType>();
+
+  if (!selfType.hasDtype() || !otherType.hasDtype()) {
+    return success();
+  }
+
+  Type selfDtype = selfType.getDtype();
+  Type otherDtype = otherType.getDtype();
+
+  auto selfRankedType = getSelf().getType().cast<RankedTensorType>();
+  auto otherRankedType = getOther().getType().cast<RankedTensorType>();
+
+  auto selfShape = selfRankedType.getShape();
+  auto otherShape = otherRankedType.getShape();
+
+  int64_t selfRank = selfRankedType.getRank();
+  int64_t otherRank = otherRankedType.getRank();
+
+  // check if both input tensors have the same number of dims
+  if (selfRank != otherRank) {
+    return emitOpError("input tensors must have the same number of dimensions, "
+                       "but got ")
+           << selfRank << " and " << otherRank;
+  }
+
+  int64_t dim;
+  if (!matchPattern(getDim(), m_TorchConstantInt(&dim))) {
+    return emitOpError("dim must be a constant int");
+  }
```

Comment:
nit: this is not part of the semantics of the op, so the verifier should not produce an error here. Normally we check for this in conversion patterns because it either simplifies the pattern, or because the compile-time information is necessary to lower to the target backend dialect. However, it is valid for ops to have non-constant arguments (sometimes they might get folded away).

One option is to return success if we are unable to get the constant value. This way you're able to keep all the static checks below